infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/provision/UniqueNamesProvisionerTest.java

```diff
@@ -0,0 +1,104 @@
+import static org.eclipse.che.workspace.infrastructure.openshift.provision.UniqueNamesProvisioner.ROUTE_PREFIX;
+import static org.eclipse.che.workspace.infrastructure.openshift.provision.UniqueNamesProvisioner.ROUTE_SUFFIX_SIZE;
+import static org.eclipse.che.workspace.infrastructure.openshift.provision.UniqueNamesProvisioner.SEPARATOR;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodBuilder;
+import io.fabric8.openshift.api.model.Route;
+import io.fabric8.openshift.api.model.RouteBuilder;
+import java.util.HashMap;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InternalEnvironment;
+import org.eclipse.che.workspace.infrastructure.openshift.environment.OpenShiftEnvironment;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+/**
+ * Tests {@link UniqueNamesProvisioner}.
+ *
+ * @author Anton Korneta
+ */
+@Listeners(MockitoTestNGListener.class)
+public class UniqueNamesProvisionerTest {
+
+  private static final String WORKSPACE_ID = "workspace37";
+  private static final String POD_NAME = "testPod";
+  private static final String ROUTE_NAME = "testRoute";
+
+  @Mock private InternalEnvironment environment;
+  @Mock private OpenShiftEnvironment osEnv;
+  @Mock private RuntimeIdentity runtimeIdentity;
+
+  private UniqueNamesProvisioner uniqueNamesProvisioner;
+
+  @BeforeMethod
+  public void setup() {
+    uniqueNamesProvisioner = new UniqueNamesProvisioner();
+  }
+
+  @Test
+  public void provideUniquePodsNames() throws Exception {
+    when(runtimeIdentity.getWorkspaceId()).thenReturn(WORKSPACE_ID);
+    final HashMap<String, Pod> pods = new HashMap<>();
+    pods.put(POD_NAME, newPod());
+    doReturn(pods).when(osEnv).getPods();
+
+    uniqueNamesProvisioner.provision(environment, osEnv, runtimeIdentity);
+
+    final String expected = WORKSPACE_ID + SEPARATOR + POD_NAME;
+    assertEquals(osEnv.getPods().get(expected).getMetadata().getName(), expected);
+  }
+
+  @Test
+  public void provideUniqueRoutesNames() throws Exception {
+    final HashMap<String, Route> routes = new HashMap<>();
+    routes.put(POD_NAME, newRoute());
+    doReturn(routes).when(osEnv).getRoutes();
+
+    uniqueNamesProvisioner.provision(environment, osEnv, runtimeIdentity);
+
+    final String actual =
+        osEnv
+            .getRoutes()
+            .values()
+            .stream()
+            .map(r -> r.getMetadata().getName())
+            .collect(toList())
+            .get(0);
+    assertTrue(actual.startsWith(ROUTE_PREFIX));
```

Comment:
You can match it by regexp instead of assuming here that prefix + length are sufficient checks and it is clear what is happening in this test

Reply:
replaced with regex